modin/core/io/column_stores/parquet_dispatcher.py

```diff
@@ -23,6 +37,317 @@
+        return complete_index, len(index_columns) == 0 or any(
+            not isinstance(c, str) for c in index_columns
+        )
+
+    @classmethod
+    def build_columns(cls, columns):
+        """
+        Split columns into chunks that should be read be workers.
+
+        Parameters
+        ----------
+        columns : list
+            List of columns that should be read from file.
+
+        Returns
+        -------
+        col_partitions : list
+            List of lists with columns for reading by workers.
+        column_widths : list
+            List with lengths of `col_partitions` subarrays
+            (number of columns that should be read by workers).
+        """
+        columns_length = len(columns)
+        if columns_length == 0:
+            return [], []
+        num_partitions = NPartitions.get()
+        column_splits = (
+            columns_length // num_partitions
+            if columns_length % num_partitions == 0
+            else columns_length // num_partitions + 1
+        )
+        col_partitions = [
+            columns[i : i + column_splits]
+            for i in range(0, columns_length, column_splits)
+        ]
+        column_widths = [len(c) for c in col_partitions]
+        return col_partitions, column_widths
+
+    @classmethod
+    def build_dtypes(cls, partition_ids, columns):
+        """
+        Compute common for all partitions `dtypes` for each of the DataFrame column.
+
+        Parameters
+        ----------
+        partition_ids : list
+            Array with references to the partitions data.
+        columns : list
+            List of columns that should be read from file.
+
+        Returns
+        -------
+        dtypes : pandas.Series
+            Series with dtypes for columns.
+        """
+        dtypes = pandas.concat(cls.materialize(list(partition_ids)), axis=0)
+        dtypes.index = columns
+        return dtypes
+
+    @classmethod
+    def build_query_compiler(cls, path, columns, **kwargs):
+        """
+        Build query compiler from deployed tasks outputs.
+
+        Parameters
+        ----------
+        path : str, path object or file-like object
+            Path to the file to read.
+        columns : list
+            List of columns that should be read from file.
+        **kwargs : dict
+            Parameters of deploying read_* function.
+
+        Returns
+        -------
+        new_query_compiler : BaseQueryCompiler
+            Query compiler with imported data for further processing.
+        """
+        col_partitions, column_widths = cls.build_columns(columns)
+        partition_ids = cls.call_deploy(path, col_partitions, **kwargs)
+        index, needs_index_sync = cls.build_index(path, kwargs["storage_options"])
+        remote_parts = cls.build_partition(partition_ids, column_widths)
+        if len(partition_ids) > 0 and len(partition_ids[0]) > 0:
+            first_row = partition_ids[0]
+            dtypes = cls.build_dtypes(
```

Comment:
what if dtypes are different in other partitions?..

Reply:
The way we are dividing up the partitions right now is by row-group, so we are splitting things horizontally. Thus, just getting the first partition should be enough. This is poorly named, and I will push some changes that reflect a better naming convention.

Reply:
This is the assumption, but we may have to revisit this.

Reply:
@vnlitvinov I think you may be right on this - I think it may not be worth even trying to compute dtypes ahead of time and instead lazily compute them when needed.